test/dynamo/test_decorators.py

```diff
@@ -490,6 +490,33 @@ def fn(x):
         res = opt_fn(x)
         self.assertEqual(ref, res)
 
+    def test_nonstrict_trace_on_method(self):
+        class Num:
+            def __init__(self, n):
+                self.n = n
+
+            @torch._dynamo.nonstrict_trace
+            def trace_me(self, t):
+                torch._dynamo.graph_break()
+                return t + self.n
```

Comment:
What's the requirement here? that `self` has to be pytree-able too?

Reply:
Yep, semantically `@torch._dynamo.nonstrict_trace` treats the method as a regular function, i.e., `Num.trace_me`, and everything else applies. If `self` is not pytree-able, we'll fail accordingly, with the error messages from #146367.